src/tracks/labeltrack/ui/LabelTrackView.cpp

```diff
@@ -1676,37 +1678,56 @@ bool LabelTrackView::DoKeyDown(
       case WXK_NUMPAD_TAB:
          if (!mLabels.empty()) {
             int len = (int) mLabels.size();
-            if (event.ShiftDown()) {
-               mSelIndex = len - 1;
-               if (newSel.t0() > mLabels[0].getT0()) {
-                  while (mSelIndex >= 0 &&
-                         mLabels[mSelIndex].getT0() > newSel.t0()) {
-                     --mSelIndex;
-                  }
-               }
-            } else {
-               mSelIndex = 0;
-               if (newSel.t0() < mLabels[len - 1].getT0()) {
-                  while (mSelIndex < len &&
-                         mLabels[mSelIndex].getT0() < newSel.t0()) {
-                     ++mSelIndex;
-                  }
-               }
+            if (IsValidIndex(mNavigationIndex, project))
+            {
+                if (event.ShiftDown()) {
+                    --mNavigationIndex;
+                }
+                else {
+                    ++mNavigationIndex;
+                }
+                mNavigationIndex = (mNavigationIndex + (int)mLabels.size()) % (int)mLabels.size();    // wrap round if necessary
+            }
+            else
+            {
+                if (event.ShiftDown()) {
+                    mNavigationIndex = len - 1;
+                    if (newSel.t0() > mLabels[0].getT0()) {
+                        while (mNavigationIndex >= 0 &&
+                            mLabels[mNavigationIndex].getT0() > newSel.t0()) {
+                            --mNavigationIndex;
+                        }
+                    }
+                }
+                else {
+                    mNavigationIndex = 0;
+                    if (newSel.t0() < mLabels[len - 1].getT0()) {
+                        while (mNavigationIndex < len &&
+                            mLabels[mNavigationIndex].getT0() < newSel.t0()) {
+                            ++mNavigationIndex;
+                        }
+                    }
+                }
```

Comment:
More is happening here than the routine changes needed elsewhere for the distinction of the two indices.  If it were me, I might make two commits.

Rather than that, add more explanation to the commit comment about the intention.